crates/p2p/src/swarm/mod.rs

```diff
@@ -56,16 +56,42 @@ static TOPIC: LazyLock<Sha256Topic> = LazyLock::new(|| Sha256Topic::new("bitvm2"
 /// Global name of the protocol
 const PROTOCOL_NAME: &str = "/strata-bitvm2";
 
-#[derive(Snafu, Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
-    #[snafu(display("Database error: {source}"))]
-    Repository { source: RepositoryError },
-
-    #[snafu(whatever, display("{message}"))]
-    Whatever {
-        message: String,
-        #[snafu(source(from(Box<dyn std::error::Error>, Some)))]
-        source: Option<Box<dyn std::error::Error>>,
+    #[error("Database error")]
+    Repository(#[from] RepositoryError),
+    #[error("Validation error")]
+    Validation(#[from] ValidationError),
+    #[error("Protocol error")]
+    Protocol(#[from] ProtocolError),
+}
+
+#[derive(Debug, Error)]
+pub enum ValidationError {
+    #[error("Invalid signature")]
+    InvalidSignature,
+    #[error("Not in signers allowlist")]
+    NotInSignersAllowlist,
+}
+
+#[derive(Debug, Error)]
+pub enum ProtocolError {
+    #[error("Failed to listen: {0}")]
+    Listen(#[from] TransportError<io::Error>),
+    #[error("Events channel closed: {source}")]
+    EventsChannelClosed {
+        #[source]
+        source: Box<dyn std::error::Error>,
+    },
+    #[error("Failed to initialize transport: {source}")]
+    TransportInitialization {
+        #[source]
+        source: Box<dyn std::error::Error>,
+    },
+    #[error("Failed to initialize behaviour: {source}")]
+    BehaviourInitialization {
+        #[source]
+        source: Box<dyn std::error::Error>,
     },
```

Comment:
Maybe let's make them `BehaviourInitialization(Box<dyn std::error::Error>)` too? Don't see a reason to use `source` here